src/pyodide.js

```diff
@@ -176,7 +175,19 @@ var languagePluginLoader = new Promise((resolve, reject) => {
         } else {
           script.src = `${package_uri}`;
         }
-        script.onerror = (e) => { reject(e); };
+        script.onerror = (e) => {
+          // If the package_uri fails to load, call monitorRunDependencies twice
+          // (so packageCounter will still hit 0 and finish loading), and remove
+          // the package from toLoad so we don't mark it as loaded.
+          console.log(`Couldn't load package from URL ${script.src}`)
+          let index = toLoad.indexOf(package);
+          if (index !== -1) {
+            toLoad.splice(index, 1);
+          }
+          for (let i = 0; i < 2; i++) {
+            window.pyodide._module.monitorRunDependencies();
+          }
+        };
```

Comment:
I think this seems fine and it should work. I will still test it out manually by running ``./bin/pyodide serve`` and let you know. However, 
- a) Shouldn't ``console.log`` be ``console.error`` instead? 

- b) Note that we might end up using ``window.onerror`` too as seen in the PR for ABI numbers here: https://github.com/iodide-project/pyodide/pull/286 

[specifically, ``https://github.com/iodide-project/pyodide/blob/43339d9880387a09f73e0afd139ea8a400107480/src/pyodide.js#L180``] 

so as to avoid waiting forever for the promise to be fulfilled, do we also use the same snippet as used here inside ``window.onerror`` too? If yes, maybe we could create a separate function for the snippet and call it inside both ``script.onerror`` and ``window.onerror``?

What do you think?

Reply:
> a) Shouldn't console.log be console.error instead?

Sure.  That makes sense.

> b) Note that we might end up using window.onerror too as seen in the PR for ABI numbers here: #286

That would have us handling every error that the window could throw (and there are a lot of them) as if it was a package failing to load.  I don't think that makes sense.

> so as to avoid waiting forever for the promise to be fulfilled, do we also use the same snippet as used here inside window.onerror too?

It would be up to the caller to handle the promise timing out and deciding what to do.  Which this change, if a package fails to load (because the URL is a 404, or the URL times out) or if it loads but there's an error in it, the counter will tick down and the promise will eventually finish.

But maybe I'm misunderstanding you.  Could you be more specific?

Reply:
This is probably not urgent right now and we can address it in the PR for ABI numbers https://github.com/iodide-project/pyodide/pull/286 if at all the need be.